crates/markdown_preview/src/markdown_preview_view.rs

```diff
@@ -38,29 +44,120 @@ impl MarkdownPreviewView {
 
     pub fn new(
         active_editor: View<Editor>,
-        languages: Arc<LanguageRegistry>,
-        cx: &mut ViewContext<Self>,
-    ) -> Self {
-        let focus_handle = cx.focus_handle();
-
-        cx.subscribe(&active_editor, |this, editor, event: &EditorEvent, cx| {
-            if *event == EditorEvent::Edited {
-                let editor = editor.read(cx);
-                let contents = editor.buffer().read(cx).snapshot(cx).text();
-                this.contents = contents;
-                cx.notify();
+        workspace: WeakView<Workspace>,
+        cx: &mut ViewContext<Workspace>,
+    ) -> View<Self> {
+        cx.new_view(|cx: &mut ViewContext<Self>| {
+            let view = cx.view().downgrade();
+            let editor = active_editor.read(cx);
+
+            let file_location = MarkdownPreviewView::get_folder_for_active_editor(editor, cx);
+            let contents = editor.buffer().read(cx).snapshot(cx).text();
+            let contents = parse_markdown(&contents, file_location);
+
+            cx.subscribe(&active_editor, |this, editor, event: &EditorEvent, cx| {
+                match event {
+                    EditorEvent::Edited => {
+                        let editor = editor.read(cx);
+                        let contents = editor.buffer().read(cx).snapshot(cx).text();
+                        let file_location =
+                            MarkdownPreviewView::get_folder_for_active_editor(editor, cx);
+                        this.contents = parse_markdown(&contents, file_location);
+                        this.list_state.reset(this.contents.children.len());
+                        cx.notify();
+
+                        // TODO: This does not work as expected.
```

Comment:
If anyone has ideas on how to solve this, I'd appreciate the help!

Reply:
Would it work if cx.notify() is dropped on line 67? (I do not know for sure how notify() works, I just expect it to send whatever updates are pending)

Reply:
Unfortunately not :/